test/example/module-splitting.cpp

```diff
@@ -0,0 +1,208 @@
+#include <cassert>
+#include <iostream>
+
+#include "ir/module-splitting.h"
+#include "ir/stack-utils.h"
+#include "wasm-features.h"
+#include "wasm-printing.h"
+#include "wasm-s-parser.h"
+#include "wasm-validator.h"
+#include "wasm.h"
+
+using namespace wasm;
+
+std::unique_ptr<Module> parse(char* module) {
+  auto wasm = std::make_unique<Module>();
+  wasm->features = FeatureSet::All;
+  try {
+    SExpressionParser parser(module);
+    Element& root = *parser.root;
+    SExpressionWasmBuilder builder(*wasm, *root[0], IRProfile::Normal);
+  } catch (ParseException& p) {
+    p.dump(std::cerr);
+    Fatal() << "error in parsing wasm text";
+  }
+  return wasm;
+}
+
+void do_test(std::set<Name> keptFuncs, std::string module) {
+  WasmValidator validator;
+  bool valid;
+
+  auto primary = parse(&module.front());
```

Comment:
Unfortunately this won't work because `parse` actually modifies the string it's parsing, but `c_str` returns `const char*`. I looked a fixing this, but it was more work than would be appropriate for this PR.